drivers/hmis/app/models/hmis/form/definition_validator.rb

```diff
@@ -172,6 +173,67 @@ def check_mutually_exclusive_attributes(document)
     link_check.call(document)
   end
 
+  def check_condition(condition, item_hash, link_id)
+    return unless condition.key?('question') # Not currently validating conditions with local constants; see below
+
+    # Find the referenced question.
+    # item_hash does not contain groups, and the referenced question could be a group (with an EXISTS condition, for instance).
+    # But for now we are only validating against pick_list_options and pick_list_reference (which groups won't have).
+    # Therefore, if item_hash doesn't contain the referenced question, just return without raising an error.
+    referenced_question = item_hash[condition['question']]
+    return if referenced_question.nil?
+    return unless referenced_question['pick_list_options'] || referenced_question['pick_list_reference']
+
+    if referenced_question['pick_list_reference']
+      valid_answer_codes = pick_list_reference_to_allowed_values[referenced_question['pick_list_reference']]
+
+      # See comments on pick_list_reference_to_allowed_values.
+      # If options is nil, this picklist falls under case #1, and we don't attempt to validate its dependent questions.
+      return if valid_answer_codes.nil?
+    else
+      valid_answer_codes = referenced_question['pick_list_options'].map { |opt| opt['code'].to_s }
+      valid_answer_group_codes = referenced_question['pick_list_options'].map { |opt| opt['group_code'].to_s }.uniq
+    end
+
+    answer_codes = condition.values_at('answer_code', 'answer_codes').flatten.compact.uniq
+    if answer_codes.any?
+      (answer_codes - valid_answer_codes).each do |code|
+        add_issue("Invalid answer code: #{code} in 'enable_when' prop of #{link_id}")
+      end
+    end
+
+    if condition.key?('answer_group_code') && valid_answer_group_codes # rubocop:disable Style/GuardClause
+      code = condition['answer_group_code']
+      add_issue("Invalid answer group code: #{code} in 'enable_when' prop of #{link_id}") unless valid_answer_group_codes.include?(code)
+    end
```

Comment:
Could we do this instead? i'm not really understanding why we have the  `&& valid_answer_group_codes` check, because I think if the condition references an `answer_group_code`, and the referenced question DOES NOT hae any answer group codes, then we _should_ present an error.

```suggestion
    elsif condition.key?('answer_group_code')
      # This condition is checking against a group code, so we need to validate that the group code is still valid for the referenced question.
      # Use safe accessor on valid_answer_group_codes because this should also display a validation error if the referenced question has no valid_answer_group_codes
      group_code = condition['answer_group_code']
      add_issue("Item '#{link_id}' has a dependency on question '#{referenced_question['link_id']}', but the dependent answer group '#{group_code}' is no longer a valid choice group for that question. Please update the dependency and try again.") unless valid_answer_group_codes&.include?(group_code)
    end
```